src/main/java/com/snowflake/kafka/connector/records/IcebergTableStreamingRecordMapper.java

```diff
@@ -54,7 +53,7 @@ private Map<String, Object> getMapForSchematization(JsonNode node) {
             entry ->
                 new AbstractMap.SimpleEntry<>(
                     Utils.quoteNameIfNeeded(entry.getKey()), entry.getValue()))
-        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        .collect(HashMap::new, (m, v) -> m.put(v.getKey(), v.getValue()), HashMap::putAll);
```

Comment:
This lin threw NPE when entry value was null. This is known JDK bug https://stackoverflow.com/questions/24630963/nullpointerexception-in-collectors-tomap-with-null-entry-values.
Record that contained `"array" : null` threw that NPE.